lib/engine-addon.js

```diff
@@ -185,6 +185,29 @@ var buildEngineJSTreeWithoutRoutes = memoize(function buildEngineJSTreeWithoutRo
   });
 });
 
+var buildEngineRoutesJSTree = memoize(function buildEngineRoutesJSTree() {
+  // Get complete engine JS tree
+  var engineAppTree = buildEngineJSTree.call(this);
+  // Separate routes
+  var engineRoutesTree = new DependencyFunnel(engineAppTree, {
+    include: true,
+    entry: this.name+'/routes.js',
+    external: ['ember-engines/routes']
+  });
+  // If babel options aren't defined, we need to transpile the modules.
+  if (!this.options || !this.options.babel) {
+    engineRoutesTree = processBabel(engineRoutesTree);
+  }
+  // Concatenate routes.js and its dependencies into a single file.
+  engineRoutesTree = concat(engineRoutesTree, {
+    allowNone: true,
+    inputFiles: ['**/*.js'],
+    outputFile: 'engines-dist/' + this.name + '/assets/routes.js'
+  });
+  // Return concatenated JS tree
+  return engineRoutesTree;
```

Comment:
Can you wrap this in a `this.debugTree` so that we can have a tad easier time at debugging when things go wrong?

Specifically, I'm suggesting:

```js
return this.debugTree(engineRoutesTree, 'routes:output');
```